api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+        String redhatToken = accountData.getRedhatToken();
+
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpGet request = new HttpGet(apiUrl);
+
+        // Set the access token in the Authorization header
+        request.setHeader(HttpHeaders.AUTHORIZATION, BEARER_PREFIX + redhatToken);
+
+        try {
+            // Make the API request
+            HttpResponse response = httpClient.execute(request);
+            // Get the HTTP status code
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            // Create an AccountValidationResponse based on the status code
+            if (statusCode == 200) {
+                accountResponse.setValidationStatus(SUCCESS);
+                accountResponse.setMessage(VALIDATION_SUCCESSFUL);
+                LOGGER.info(VALIDATION_SUCCESSFUL);
+            } else {
+                accountResponse.setValidationStatus(FAILURE);
+                accountResponse.setMessage(TOKEN_VALIDATION_FAILED);
+                accountResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE +
+                        HTTP_STATUS_CODE_TEMPLATE, VALIDATING, statusCode));
+                LOGGER.info(VALIDATION_FAILED);
+                LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE +
+                        HTTP_STATUS_CODE_TEMPLATE, VALIDATING, statusCode));
+            }
+        } catch (UnknownHostException e) {
+            accountResponse.setValidationStatus(FAILURE);
+            accountResponse.setMessage(ACCOUNT_ID_VALIDATION_FAILED);
+            accountResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+            LOGGER.info(ACCOUNT_ID_VALIDATION_FAILED);
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+        } catch (Exception e) {
+            accountResponse.setValidationStatus(FAILURE);
+            accountResponse.setMessage(VALIDATION_FAILED);
+            accountResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+            LOGGER.info(VALIDATION_FAILED);
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+        }
+        return accountResponse;
+    }
+
+    @Override
+    public AccountValidationResponse addAccount(CreateAccountRequest accountData) {
+        LOGGER.info(VALIDATING_REDHAT_DETAILS);
+        if (accountData.getRedhatAccountName() == null || accountData.getRedhatAccountName().isEmpty()) {
+            accountData.setRedhatAccountName(accountData.getRedhatAccountId());
+        }
+        AccountValidationResponse accountResponse = validateRequest(accountData);
+        if (accountResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info(VALIDATION_FAILED_DUE_TO_MISSING_PARAMETERS);
+            return accountResponse;
+        }
```

Comment:
you should do this on line 131 before you can guarantee that accountId exists. Move this as first "if"